web/docs/tql2/functions/replace.md

```diff
@@ -0,0 +1,55 @@
+# replace
+
+Replaces characters within a string.
+
+```tql
+replace(x:string, pattern:string, replacement:string, [max=int]) -> string
+```
+
+## Description
+
+The `replace` function returns a new string where occurrences of `pattern` in `x`
+are replaced with `replacement`, up to `max` times. If `max` is omitted, all
+occurrences are replaced.
+
+### `x: string`
+
+The subject to replace the action on.
+
+### `pattern: string`
+
+The pattern to replace in `x`.
+
+### `replacement: string`
+
+The replacement value for `pattern`.
+
+### `max = string` (optional)
```

Comment:
`(optional)` is inside the back-ticks on the operator pages, should be consistent here.

Reply:
Curious: we are putting a question mark after the type in the diagnostic. In the docs, we use `[key=string]` in the synopsis. We haven't found a good solution for this yet.

Reply:
Yes, the question mark was the original design, and we tried to improve it with the docs. Short-term, the diagnostics should be aligned with the docs. Long-term, we can see whether there is an even better solution.

Reply:
So how should we document this then?

```markdown
### [max=string]
```

or

```markdown
### `max = string (optional)`
```

or

```markdown
### `max = string` (optional)
```

I'm also torn about the extra spacing we have, e.g., `key: string` and `key = string` in the docs, versus `key:string` and `key=string` in the synopsis and diagnostics.

I think we should agree on one canonical form.

Reply:
What we previously committed to is the format as you can see it in the operator docs, including its spacing:
```md
### `max = string (optional)`
```
We can in the future agree to change it, but that's what it is right now.

Reply:
This makes the least sense semantically, as the "optional" suffix is not code. But for consistency I switched to this style.